integration_test/tests/power_bi_tests.rs

```diff
@@ -14,6 +14,75 @@ mod integration {
     use std::ptr::null_mut;
     use std::slice;
 
+    const BUFFER_LENGTH: SmallInt = 300;
+
+    macro_rules! test_get_info {
+        ($conn_handle:expr, $info_type: expr, $info_value_buffer_length: expr, $info_value_type: expr) => {{
+            let conn_handle = $conn_handle;
+            let info_type = $info_type;
+            let info_value_buffer_length = $info_value_buffer_length;
```

Comment:
The buffer creation seems to need to be created with a const value, which is why I have it using the const 300 value.  
I can use the `info_value_buffer_length` to check the expected `StringLengthPtr` after SQLGetinfoW is called instead.

Reply:
Got it! 
Sorry for the unclear comment. 
What I tried to say was that it would be closer to what users/tools do if you were constructing the output buffer based on the info_value_type. You'll still need to have a variable to check that the value in `StringLengthPtr` is correct.
The thing bothering me is simply that you are constructing a buffer of 600 bytes, but then pretend  it is smaller when doing the call based on the expected size of the data you want to retrieve.

So, I was just thinking that you could have a switch when you create `buffer` to either use a pointer on a Wchar buffer or a pointer on a smallint depending on the `info_value_type` and pass this real size.

No big deal though. It is still a valid and I think we have unit tests to check that if we handle the buffer length correctly. Plus, I read the spec again and if info type is not a VARCHAR, the value of BufferLength is ignored and the driver assumes the size of *InfoValuePtr is SQLUSMALLINT or SQLUINTEGER.

So... you can leave it as is 👍

Reply:
Ah ok.  Thanks for the additional context on how tools would be constructing the output buffer.  I was hung up on thinking we wanted to create a different sized buffer for all the different `info_type`s.  I will leave it as is for now.